src/index.js

```diff
@@ -52,6 +58,19 @@ export default class AutoCheckElement extends HTMLElement {
     this.setAttribute('csrf', value)
   }
 
+  get required(): boolean {
+    return this.hasAttribute('required')
+  }
+
+  set required(required: boolean) {
+    this.input.required = required
+    if (required) {
+      this.input.setCustomValidity('auto-check-loading')
+    } else {
+      this.input.setCustomValidity('')
+    }
+  }
```

Comment:
Setting the property shouldn't affect the validity because it will put it into an error state before we've interacted with the element. Styles might outline the input in red for example.

We should set the loading state when we emit the `loadstart` event, an error state for the `error` event, and clear it when emitting `load` event.